cranelift/codegen/src/dominator_tree.rs

```diff
@@ -580,9 +580,37 @@ impl DominatorTreePreorder {
     pub fn dominates(&self, a: Block, b: Block) -> bool {
         let na = &self.nodes[a];
         let nb = &self.nodes[b];
+
+        // A block is considered to dominate itself
+        if a == b {
+            return true;
+        }
+
+        // Dominance is ill defined for unreachable blocks
+        // Return false if either block is unreachable
+        if na.pre_number == 0 || nb.pre_number == 0 {
+            return false;
+        }
```

Comment:
Was this needed to address a failure otherwise? I'd prefer keeping the simple definition -- the original should also return false (I think?) if either is unreachable and the other is not, since the RHS of a `<=` will be zero.